cicd/base/app/base.yaml

```diff
@@ -119,43 +122,27 @@ Resources:
     Type: AWS::IAM::Role
     Properties:
       RoleName: !Sub
-                  - userdetails-role-${ResourceName}
+                  - ${pProductName}-role-${ResourceName}
                   - ResourceName: !If [ IsDev, !Ref pCleanBranch, !Ref pEnvironment ]
-      AssumeRolePolicyDocument:
-        Version: '2012-10-17'
-        Statement:
-          - Effect: Allow
-            Principal:
-              Federated: !Sub arn:aws:iam::${AWS::AccountId}:oidc-provider/oidc.eks.${AWS::Region}.amazonaws.com/id/E2B4E818912ED7B145C9B4AA0B8D83A8
-            Action: sts:AssumeRoleWithWebIdentity
-            Condition:
-              StringEquals:
-                oidc.eks.ap-southeast-2.amazonaws.com/id/E2B4E818912ED7B145C9B4AA0B8D83A8:sub: "system:serviceaccount:userdetails:userdetails-service-account"
-                oidc.eks.ap-southeast-2.amazonaws.com/id/E2B4E818912ED7B145C9B4AA0B8D83A8:aud: "sts.amazonaws.com"
-#      AssumeRolePolicyDocument: !Sub
-#        - |
-#          {
-#            "Version": "2012-10-17",
-#            "Statement": [
-#                {
-#                    "Effect": "Allow",
-#                    "Principal": {
-#                        "Federated": "arn:aws:iam::${AWS::AccountId}:oidc-provider/oidc.eks.${AWS::Region}.amazonaws.com/id/${pEksClusterId}"
-#                    },
-#                    "Action": "sts:AssumeRoleWithWebIdentity",
-#                    "Condition": {
-#                        "StringEquals": {
-#                            "${OidcProviderEndpoint}:sub": "system:serviceaccount:${pProductName}:${pServiceAccountName}"
-#                        }
-#                    }
-#                }
-#            ]
-#          }
-#        -
-#          pEksClusterId: !Ref pEksClusterId
-#          OidcProviderEndpoint: !Sub oidc.eks.${AWS::Region}.amazonaws.com/id/${pEksClusterId}
-#          pProductName: !Ref pProductName
-#          pServiceAccountName: !Ref pServiceAccountName
+      AssumeRolePolicyDocument: !Sub |
```

Comment:
I really tried to keep this in YAML format but I couldnt get the syntax right for the variable substitution on the StringEquals section.

Reply:
Hi Joe, I think it is an existing issue in StringEquals. This is the only workaround I was bale to find - https://github.com/aws/amazon-eks-pod-identity-webhook/issues/30